File: Programmers/nolv/17681.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

// 2진수 변환
vector<int> convertToBinary(int number, int n) {
    vector<int> answer;

    for (int i = 0; i < n; i++) answer.push_back(0);

    while (n-1 >= 0) {
        answer[n-1] = number % 2;
        number = number / 2;
        n--;
    }

    return answer;
}

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    vector<int> a1, a2;
    string s = "";
    // 2진수 변환
    for (int i = 0; i < n; i++) {
        a1 = convertToBinary(arr1[i], n);
        a2 = convertToBinary(arr2[i], n);

        for (int j = 0; j < n; j++) {
            if (a1[j] == 1 || a2[j] == 1) {
                s += "#";
            }
            else {
                s += " ";
            }
            
        }
        answer.push_back(s);

        a1.clear();
        a2.clear();
        s = "";

    }

    return answer;
}

int main() {

    vector<int> arr1 = { 9, 20, 28, 18, 11 };
    vector<int> arr2 = { 30, 1, 21, 17, 28 };
    solution(5, arr1, arr2);

    return 0;
}

// 다른 풀이는 비트연산자 쓰고 string이니까 앞에 붙이는 형식으로 했음..

File: Python/17681.py
def conv2bi(numbers):
    return [bin(number) for number in numbers]


def convbi2str(number1, number2):
    result = ''
    for num1, num2 in zip(number1, number2):
        if int(num1) + int(num2) != 0:
            result += "#"
        else:
            result += " "
    return result


def solution(n, arr1, arr2):
    answer = []
    # AND 연산하기
    temp_arr1, temp_arr2 = [], []
    for number1, number2 in zip(arr1, arr2):
        # 2진수로 바꾸기
        temp_arr1.append(format(number1, 'b'))
        temp_arr2.append(format(number2, 'b'))
    # fixed length
    max_length = 0
    for number1, number2 in zip(temp_arr1, temp_arr2):
        max_length = max(len(number1), len(number2), max_length)
    temp_arr1, temp_arr2 = [], []
    for number1, number2 in zip(arr1, arr2):
        temp_arr1.append(f'{number1:0{max_length}b}')
        temp_arr2.append(f'{number2:0{max_length}b}')
    # AND
    for number1, number2 in zip(temp_arr1, temp_arr2):
        answer.append(convbi2str(number1, number2))
    # # 과 공백으로 대체하기
    return answer